src/roadmap/common.py

```diff
@@ -0,0 +1,93 @@
+import json
+import logging
+import typing as t
+import urllib.parse
+import urllib.request
+
+from pathlib import Path
+from urllib.error import HTTPError
+
+from fastapi import HTTPException
+
+from roadmap.config import SETTINGS
+
+
+logger = logging.getLogger("uvicorn.error")
+
+
+# FIXME: This should be cached
+async def get_system_count_from_inventory(
+    headers: dict[str, str | None],
+    page: int = 1,
+    per_page: int = 100,
+    major: int | None = None,
+    minor: int | None = None,
+) -> dict[str, t.Any]:
+    if SETTINGS.dev:
```

Comment:
maybe break the contents of this if block into its own function for readability? It will make the "return" more obvious so folks know execution doesn't continue past the block. I had some trouble reading it